packages/loopring_v3/circuit/Gadgets/StorageGadgets.h

```diff
@@ -199,47 +213,54 @@ class StorageReaderGadget : public GadgetT {
   }
 };
 
-class NonceGadget : public GadgetT {
+// QUESTION(daniel): it seems this gadget is the same as ReadStorageGadget
+// except that it requires data to be 0.
+//
+// Maybe we should move this to `AccountGadgets.h' as this is on the account
+// level?
```

Comment:
It's not on the account level, these are on the level of the token balances so seems like putting this here makes the most sense? Because like you said above, it's just a small wrapper around `ReadStorageGadget` that only supports `0 -> 1`.

Reply:
Brecht, I didn't know we also have an *balance nonce*, when will this once go from 0 to 1 and is it possible to go from 1 back 0 later?

Reply:
I mean the nonce system used here by the transfers that is built on top of the storage tree in `ReadNonceGadget`. This storage tree is stored separately for each token, not for each account.

Reply:
I see. I would suggest to rename it to TransferStorageID or something like it.

Reply:
If the other gadget is named `ReadStorageGadget` then `ReadNonceGadget` seems fine with me to emphasise the nonce behaviour. Or maybe `ReadStorageNonceGadget` or similar. `TransferStorageID` makes me think it's a special storageID for transfers.